.github/workflows/release.yml

```diff
@@ -0,0 +1,112 @@
+# Create a new GitHub release on pushed tag stating with 'v'.
+# Run tests, build and uploads binaries.
+
+name: Release
+
+on:
+  push:
+    tags:
+    - 'v*'
+
+jobs:
+
+  test:
+    name: Release Test
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: set up Go 1.13
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+
+    - name: check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+
+    - name: sanity-check
+      run: make test
+
+  # See https://github.com/actions/upload-release-asset
+  build:
+    name: Build and Release
+    needs: test
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: set up Go 1.13
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+
+    - name: check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
+            curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+            dep ensure
+        fi
+
+    - name: parse version string
+      id: parse_version
+      run: |
+        export BUILD_VERSION=$(echo $GITHUB_REF | sed -e 's/refs.tags.\(.*\)$/\1/')
+        export BUILD_NAME=terraform-provider-rollbar_$BUILD_VERSION
+        echo "::set-output name=BUILD_NAME::$BUILD_NAME"
+        echo ::set-output name=ZIP_NAME_LINUX::"$BUILD_NAME"_linux_amd64.zip
+        echo ::set-output name=ZIP_NAME_DARWIN::"$BUILD_NAME"_darwin_amd64.zip
+
+    - name: build Linux
+      run: |
+        make build-linux
+        mv terraform-provider-rollbar ${{ steps.parse_version.outputs.BUILD_NAME }}
+        zip --junk-paths ${{ steps.parse_version.outputs.ZIP_NAME_LINUX }} ${{ steps.parse_version.outputs.BUILD_NAME }}
+
+    - name: build Darwin
```

Comment:
this sounds like a candidate for a reusable GitHub Action (generic for customizable platform)